rs/execution_environment/src/canister_manager.rs

```diff
@@ -67,6 +67,11 @@ pub(crate) mod types;
 /// Maximum binary slice size allowed per single message download.
 const MAX_SLICE_SIZE_BYTES: u64 = 2_000_000;
 
+struct ValidatedMemoryUsage {
```

Comment:
Perhaps we should leave a doc comment for this struct to let the reader know what to expect.